### PR TITLE
DSND-734: Company Profile API Happy Path Patch endpoint

### PR DESCRIPTION
**Pull request by sameerinfo2003**
DSND-734: Company Profile API Happy Path Patch endpoint

DSND-734: Company Profile API Error/Retry Integration Test Scenarios

https://companieshouse.atlassian.net/browse/DSND-734

### CODE REVIEW

File: src/itest/resources/features/company-profile-error-retry.feature [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Examples:
+      | data     |
+      | 00006400 |
+
+  Scenario Outline: Processing company profile information while database is down
+
+    Given Company profile api service is running
+    And the company profile database is down
+    When I send PATCH request with payload "<data>" and company number "<data>"
+    Then I should receive 500 status code
```
Comment (gseregni-ch) [inline src/itest/resources/features/company-profile-error-retry.feature:30]: This should return 503. Please see scenario 1 description below. 
Scenario 1 (503 - Service Unavailable):
Given Company Profile API component is successfully running but MongoDB is not reachable
When ‘PATCH’ Rest endpoint is invoked with a valid json payload but Repository throws an error (replicate MongoDB not reachable). NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 503 (Service Unavailable) to the client and data is not updated into Mongo DB, and CHS Kafka API is never invoked

File: src/itest/resources/features/company-profile-error-retry.feature [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    And the CHS Kafka API is not invoked
+
+    Examples:
+      | data     |
+      | 11748564 |
+
+  Scenario Outline: Processing company profile information unsuccessfully but saves to database
```
Comment (gseregni-ch) [inline src/itest/resources/features/company-profile-error-retry.feature:37]: The 503 should be returned by the CHS Kafka API. Please see scenario below.

Scenario 4 (503 - Service Unavailable):
Given Company Profile API component is successfully running And Stubbed CHS Kafka API endpoint will return 503 http response code 
When ‘PATCH’ Rest endpoint is invoked with a valid json payload NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 503 (Service Unavailable) to the client and data is not updated into Mongo DB

File: src/itest/resources/features/company-profile-error-retry.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
```
Comment (gseregni-ch) [inline src/itest/resources/features/company-profile-error-retry.feature:38]: The scenario below is missing.

Scenario 5 (503 - Service Unavailable):
Given Company Profile API component is successfully running but MongoDB is not reachable
When ‘GET’ Rest endpoint is invoked with a valid company number but Repository throws an error (replicate MongoDB not reachable). NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 503 (Service Unavailable) to the client

File: src/itest/resources/features/company-profile-error-retry.feature [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  Scenario Outline: Retrieve company profile unsuccessfully as the company number does not exist
+
+    Given Company profile api service is running
+    When I send GET request with company number "<data>"
+    Then I should receive 404 status code
+
+    Examples:
+      | data     |
+      | 00006400 |
+
```
Comment (gseregni-ch) [inline src/itest/resources/features/company-profile-error-retry.feature:24]: Scenario 3 is missing.

Scenario 3 (500 - Internal Server Error):
Given Company Profile API component is successfully running 
When ‘PATCH’ Rest endpoint is invoked with a valid json payload that causes a NPE (make some attribute in payload as null) NOTE: Please reuse valid inputs from Happy Path integration story for same component)
Then Rest endpoint returns http response code 500 (Internal Error) to the client and data is not updated into Mongo DB, and CHS Kafka API is never invoked

**Review by gseregni-ch — CHANGES_REQUESTED** [review]

File: src/itest/resources/features/company-profile-error-retry.feature (hunk 4 above)
Comment (sameerinfo2003) [reply src/itest/resources/features/company-profile-error-retry.feature:36]: Changes done

File: src/itest/resources/features/company-profile-error-retry.feature (hunk 2 above)
Comment (sameerinfo2003) [reply src/itest/resources/features/company-profile-error-retry.feature:60]: Changes done

File: src/itest/resources/features/company-profile-error-retry.feature (hunk 3 above)
Comment (sameerinfo2003) [reply src/itest/resources/features/company-profile-error-retry.feature:61]: Changes done

**Review by gseregni-ch — APPROVED** [review]

**Pull request merged**